spring-boot-project/spring-boot/src/main/java/org/springframework/boot/DigitalAmount.java

```diff
@@ -0,0 +1,307 @@
+	 * @return {@code this - other}
+	 */
+	public DigitalAmount subtract(long amount, @NonNull DigitalUnit unit) {
+		return subtract(fromUnit(amount, unit));
+	}
+
+	/**
+	 * Returns a new {@code DigitalAmount} whose value is {@code (this +
+	 * other)}.
+	 * @param other value to be added to this {@code DigitalAmount}.
+	 * @return {@code this + other}
+	 */
+	public DigitalAmount add(@NonNull DigitalAmount other) {
+		Assert.notNull(other, () -> "Digital Amount must not be null");
+		return new DigitalAmount(Math.addExact(this.amount, other.amount));
+	}
+
+	/**
+	 * Returns a new {@code DigitalAmount} whose value is {@code (this -
+	 * other)}.
+	 * @param other value to be subtracted from this {@code DigitalAmount}.
+	 * @return {@code this - other}
+	 */
+	public DigitalAmount subtract(@NonNull DigitalAmount other) {
+		Assert.notNull(other, () -> "Digital Amount must not be null");
+		return new DigitalAmount(Math.subtractExact(this.amount, other.amount));
+	}
+
+	@Override
+	public int compareTo(@NonNull DigitalAmount other) {
+		Assert.notNull(other, () -> "Digital Amount must not be null");
+		return Long.compare(this.amount, other.amount);
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		DigitalAmount that = (DigitalAmount) o;
+		return this.amount == that.amount;
+	}
+
+	@Override
+	public int hashCode() {
+		return Long.hashCode(this.amount);
+	}
+
+	/**
+	 * Print this {@code DigitalAmount} using the given unit.
+	 * @param unit the {@code DigitalUnit} to use if the value doesn't specify,
+	 * {@link DigitalUnit#BYTES} will be used
+	 * @return the printed result
+	 * @see DigitalAmountStyle
+	 */
+	public String print(@Nullable DigitalUnit unit) {
+		return DigitalAmountStyle.SIMPLE.print(this, unit);
+	}
+
+	@Override
+	public String toString() {
+		return print(DigitalUnit.BYTES);
+	}
+
+	/**
+	 * Parse the given value to a {@code DigitalAmount} using
+	 * {@link DigitalAmountStyle#SIMPLE}.
+	 * @param value the value to parse
+	 * @return a {@code DigitalAmount}
+	 * @see #parse(CharSequence, DigitalUnit)
+	 */
+	public static DigitalAmount parse(@NonNull CharSequence value) {
+		return parse(value, DigitalUnit.BYTES);
+	}
+
+	/**
+	 * Parse the given value to a {@code DigitalAmount} using
+	 * {@link DigitalAmountStyle#SIMPLE}.
+	 * @param value the value to parse
+	 * @param unit {@code DigitalUnit} to use if one is not specified, default value is
+	 * {@link DigitalUnit#BYTES}
+	 * @return a {@code DigitalAmount}
+	 */
+	public static DigitalAmount parse(@NonNull CharSequence value,
+			@Nullable DigitalUnit unit) {
+		return DigitalAmountStyle.SIMPLE.parse(value, unit);
+	}
+
+	/**
+	 * Creates a new {@code DigitalAmount} from the bytes.
+	 * @param amount amount of the bytes
+	 * @return a {@code DigitalAmount}
+	 */
+	public static DigitalAmount fromBytes(long amount) {
```

Comment:
I'd use `of` to be consistent with `Duration` and have a value with a unit rather than those dedicated methods. 

The surface area of this object has to remain small given where it is actually going to be used.